applications/ConstitutiveLawsApplication/custom_constitutive/traction_separation_law.cpp

```diff
@@ -0,0 +1,751 @@
+        }
+
+        const double tolerance = std::numeric_limits<double>::epsilon();
+        Vector DelaminationDamageModeOne(p_constitutive_law_vector.size()+1);
+        Vector DelaminationDamageModeTwo(p_constitutive_law_vector.size()+1);
+        Vector ThresholdModeOne(p_constitutive_law_vector.size()-1);
+        Vector ThresholdModeTwo(p_constitutive_law_vector.size()-1);
+    
+        noalias(DelaminationDamageModeOne) = mDelaminationDamageModeOne;
+        noalias(DelaminationDamageModeTwo) = mDelaminationDamageModeTwo;
+        noalias(ThresholdModeOne) = mThresholdModeOne;
+        noalias(ThresholdModeTwo) = mThresholdModeTwo;
+
+
+        for(IndexType i=0; i < p_constitutive_law_vector.size()-1; ++i) {
+
+            interfacial_stress[i][0] = AdvancedConstitutiveLawUtilities<VoigtSize>::MacaullyBrackets((layer_stress[i][2] + layer_stress[i+1][2]) * 0.5); // interfacial normal stress
+            interfacial_stress[i][1] = (layer_stress[i][4] + layer_stress[i+1][4]) * 0.5; // interfacial shear stress
+            interfacial_stress[i][2] = (layer_stress[i][5] + layer_stress[i+1][5]) * 0.5; // interfacial shear stress
+
+            const double equivalent_stress_mode_one = interfacial_stress[i][0];
+            const double equivalent_stress_mode_two = std::sqrt(std::pow(interfacial_stress[i][1],2.0)+std::pow(interfacial_stress[i][2],2.0));
+
+            if ((layer_stress[i][2] + layer_stress[i+1][2] * 0.5) < tolerance) {
+                negative_interfacial_stress_indicator[i+1] = true;
+            }
+
+            // Damage calculation
+
+            const double T0n = r_material_properties[INTERFACIAL_NORMAL_STRENGTH]; // Interfacial Normal Strength
+            const double T0s = r_material_properties[INTERFACIAL_SHEAR_STRENGTH]; // Interfacial Shear Strength
+            const double GIc = r_material_properties[MODE_ONE_FRACTURE_ENERGY]; // Mode I Energy Release Rate
+            const double GIIc = r_material_properties[MODE_TWO_FRACTURE_ENERGY]; // Mode II Energy Release Rate
+            const double Ei = r_material_properties[TENSILE_INTERFACE_MODULUS]; // Tensile modulus of the interface
+            const double Gi = r_material_properties[SHEAR_INTERFACE_MODULUS]; // Shear modulus of the interface
+            const double characteristic_length = 0.6343 * (AdvancedConstitutiveLawUtilities<VoigtSize>::CalculateCharacteristicLengthOnReferenceConfiguration(rValues.GetElementGeometry()));
+        
+            const double F_mode_one = equivalent_stress_mode_one - ThresholdModeOne[i];
+            if (F_mode_one > tolerance) {
+
+                // const double AParameter_mode_one = -std::pow(T0n, 2) / (2.0 * Ei * GIc / characteristic_length); // Linear
+                const double AParameter_mode_one = 1.00 / (GIc * Ei / (characteristic_length * std::pow(T0n, 2)) - 0.5); // Exponential
+                
+                KRATOS_ERROR_IF(AParameter_mode_one < 0.0) << "AParameter_mode_one is negative." << std::endl;
```

Comment:
Can you do this check in the `InitializeMaterial` too? in this way you can know if `Gf` is too low at the very beginning of the simulation and not only when damage starts.

Reply:
hence saving time and money in psychologists

Reply:
Done!